File: OOP/OOP_variable/oop_value.py
class C(object):
    # 외부에서 접근 가능한 변수 = 속성(property)
    def __init__(self, v):
        self.__value = v 
        # value에 __를 하면 외부에서 읽을 수 없는 인스턴스 변수가 된다.

    def show(self):
        print(self.__value)


c1 = C(10)
#print(c1.__value)
c1.show() # 메서드 안에서는 읽을 수 없는 인스턴스도 읽을 수 있다.

File: OOP/OOP_variable/oop_value.rb
class C
  # 외부에서 접근 가능한 변수 = 속성 (attribute)
  # attr_reader :value # value를 읽기 가능한 속성으로 지정
  # attr_writer :value # value를 쓰기 가능한 속성으로 지정
  attr_accessor :value # value를 읽기/쓰기 둘다 가능한 속성으로 지정
  def initialize(v)
    @value = v
  end
  def show()
    p @value
  end
end
c1 = C.new(10)
p c1.value
c1.value = 20
p c1.value
